swap/swap_test.go

```diff
@@ -1159,10 +1223,20 @@ func testDeploy(ctx context.Context, swap *Swap) (err error) {
 	opts.Value = big.NewInt(42)
 	opts.Context = ctx
 
-	swap.contract, _, err = cswap.Deploy(opts, swap.backend, swap.owner.address, defaultHarddepositTimeoutDuration)
+	factory, err := cswap.FactoryAt(testBackend.factoryAddress, testBackend)
```

Comment:
You don't have to create a new factory here. Factory is already part of `swap`

Reply:
fixed